make older vs2015_runtime conflict with ucrt

isuruf opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

isuruf commented:
cc @xhochy.

After this, I'm going to make a new package `vs2015_runtime` that depend on ucrt.

isuruf commented:
ucrt is a new package that we added for vs2019 update 9's `vs2015_runtime`. Before that `ucrt` was part of `vs2015_runtime`.
To avoid mixing `ucrt` and `vs2015_runtime` we added a `run_constrained` on `ucrt` to require the new `vs2015_runtime`.
Issue with this was that, conda-build gives the following warning,
```
Needed DSO api-ms-win-crt-string-l1-1-0.dll found in ['ucrt']
but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

To avoid that we need to add a `run_exports` for `ucrt` for `vs2017_win-64`, but `vs2017_win-64` should only require VS 2017 runtime, but that conflicts with `ucrt`. So, what I want to do is to merge this PR, remove the `run_constrained` on the newer runtime from `ucrt` (which is still constrained by the `run_constrained` on ucrt from older runtimes) and then make a new runtime package that depends on ucrt even for VS 2017.
